yaks/lib/yaks/reader/hal.rb

```diff
@@ -29,7 +29,7 @@ def convert_links(links)
             rel = rel.to_sym if rel =~ /\A\w+\z/
             Resource::Link.new(rel: rel, uri: l['href'], options: options)
           end
-        end.to_set
+        end
```

Comment:
To some extent, I don't see how it would make sense to have two identical links. But the ordering is something I've actually ran into.